spark/src/main/scala/io/delta/tables/execution/VacuumTableCommand.scala

```diff
@@ -53,17 +56,33 @@ case class VacuumTableCommand(
         "VACUUM",
         DeltaTableIdentifier(path = Some(deltaTable.path.toString)))
     }
-    VacuumCommand.gc(sparkSession, deltaTable.deltaLog, dryRun, horizonHours).collect()
+    val reservoir = reservoirTable match {
+      case Some(table) => Some(
+        getDeltaTable(sparkSession.sessionState.analyzer.execute(table), "VACUUM")
+          .toDf(sparkSession)
+      )
+      case None => reservoirQuery.map(sparkSession.sql)
+    }
```

Comment:
Equivalent but shorter?
```suggestion
    val reservoir = reservoirTable.map(table =>
        getDeltaTable(sparkSession.sessionState.analyzer.execute(table), "VACUUM")
          .toDf(sparkSession)
      ).getOrElse(reservoirQuery.map(sparkSession.sql))
```